datumaro/plugins/searcher.py

```diff
@@ -0,0 +1,171 @@
+    return image
+
+
+def tokenize(texts: str, context_length: int = 77):
+    checkpoint = "openai/clip-vit-base-patch32"
+    tokenizer = CLIPTokenizer.from_pretrained(checkpoint)
+    eot_token = tokenizer.encoder["<|endoftext|>"]
+    tokens = tokenizer.encode(texts)
+
+    result = np.zeros((1, context_length))
+    if len(result) > context_length:
+        result = result[:context_length]
+        result[-1] = eot_token
+
+    for i, token in enumerate(tokens):
+        result[:, i] = token
+    return result
+
+
+class SearcherLauncher(OpenvinoLauncher):
+    def __init__(
+        self,
+        description=None,
+        weights=None,
+        interpreter=None,
+        device=None,
+        model_dir=None,
+        output_layers=None,
+    ):
+        model_name = "clip_visual_ViT-B_32"
+        if description or weights:
+            model_name = osp.splitext(description)[0]
+
+        url_folder = "http://s3.toolbox.iotg.sclab.intel.com/test/components-data/datumaro/models/"
+        if not model_dir:
+            model_dir = os.path.join(os.path.expanduser("~"), ".cache/datumaro")
+            if not osp.exists(model_dir):
+                os.mkdir(model_dir)
+
+        description = osp.join(model_dir, model_name + ".xml")
+        if not osp.exists(description):
+            cached_description_url = osp.join(url_folder, model_name + ".xml")
+            log.info('Downloading: "{}" to {}\n'.format(cached_description_url, description))
+            download_file(cached_description_url, description)
+
+        weights = osp.join(model_dir, model_name + ".bin")
+        if not osp.exists(weights):
+            cached_weights_url = osp.join(url_folder, model_name + ".bin")
+            log.info('Downloading: "{}" to {}\n'.format(cached_weights_url, weights))
+            download_file(cached_weights_url, weights)
+
+        if not interpreter:
+            openvino_plugin_samples_dir = "datumaro/plugins/openvino_plugin/samples"
+            interpreter = osp.join(openvino_plugin_samples_dir, "clip_ViT-B_32_interp.py")
+
+        super().__init__(description, weights, interpreter, device, model_dir, output_layers)
+
+        self._device = device or "cpu"
+        self._output_blobs = next(iter(self._net.outputs))
+        self._input_blobs = next(iter(self._net.input_info))
+
+    def infer(self, inputs):
+        if isinstance(inputs, str):
+            if len(inputs.split()) > 1:
+                prompt_text = inputs
+            else:
+                prompt_text = f"a photo of a {inputs}"
+            inputs = tokenize(prompt_text)
+        else:
+            inputs = inputs.squeeze()
+            if not inputs.any():
+                # media.data is None case
+                return None
+
+            if np.array(inputs).ndim == 2:
+                inputs = cv2.cvtColor(inputs, cv2.COLOR_GRAY2RGB)
+            elif np.array(inputs).ndim == 4:
+                inputs = cv2.cvtColor(inputs, cv2.COLOR_RGBA2RGB)
```

Comment:
Datumaro uses `BGR` format. 
https://github.com/openvinotoolkit/datumaro/blob/3aa474226f19cee00f2e8e78347d8a11a4c4ff8a/datumaro/util/image.py#L59-L86
If the model requires `RGB` format. You might also convert it when the number of channels is three.

Reply:
Fix. Thank you for reminding me this.